mongoexport/main/mongoexport.go

```diff
@@ -65,23 +64,6 @@ func main() {
 	// verify uri options and log them
 	opts.URI.LogUnsupportedOptions()
 
-	provider, err := db.NewSessionProvider(*opts)
-	if err != nil {
-		log.Logvf(log.Always, "%v", err)
-		os.Exit(util.ExitError)
-	}
-	defer provider.Close()
-
-	// temporarily allow secondary reads for the isMongos check
-	provider.SetReadPreference(mgo.Nearest)
```

Comment:
This is already taken care of by isMongos (https://github.com/mongodb/mongo-tools-common/blob/master/db/command.go#L125)